app/main.py

```diff
@@ -1,2 +1,134 @@
+BLANK = object()
+
+
 class Dictionary:
-    pass
+    def __init__(self):
+        self._hash_table_size = 8
+        self._cells = [None] * self._hash_table_size
+        self._length = 0
+
+    @property
+    def table(self):
+        table = []
+        for cell in self._cells:
+            if cell not in (None, BLANK):
+                key, _, value = cell
+                table.append((key, value))
+
+        return table
+
+    def __setitem__(self, key, value):
+        if self._is_necessary_hash_table_resizing():
+            self._perform_hash_table_resize()
+
+        self._fill_hash_table_cell(self._cells, key, value)
+
+    def __getitem__(self, item):
+        hashed_index = self._get_hashed_index(item)
+
+        while self._cells[hashed_index] is not None:
+            if self._cells[hashed_index] is not BLANK:
+                cell_key, cell_hash, cell_value = self._cells[hashed_index]
+                if cell_key == item:
+                    return cell_value
+                hashed_index = (hashed_index + 1) % self._hash_table_size
+        else:
+            raise KeyError
+
+    def __len__(self):
+        return self._length
+
+    def _get_hashed_index(self, key):
+        hashed_key = hash(key)
+        index = hashed_key % self._hash_table_size
+        return index
+
+    def _is_necessary_hash_table_resizing(self):
+        threshold = int(2 / 3 * self._hash_table_size)
+        if self._length >= threshold:
+            self._hash_table_size *= 2
```

Comment:
Shouldn't this action be placed in the `_perform_hash_table_resize ` method? 
I believe it's a part of that logic